### PR TITLE
Draft code for review: capturing votable metadata

### LINKED ISSUE #7759: astropy.io.votable and hence astropy.table does not support global VOTable metadata
The VOTable standard provides for the inclusion of global metadata via <PARAM>, <GROUP>, and <INFO> tags.  This is an important use for VOTables, where such metadata can provide the framework within which the table data is interpreted - sers of FITS tables do this all the time.  astropy.votable has Param,  Group, and Info classes but these are not used for global metadata  (these tags can also appear inside the data portion of a table).  Such metadata is neither parsed in input nor written during output of a VOTable.

This serious failure should be noted in the documentation.  Someone sufficiently familiar with the innards of astropy.io.votable could easily include this capability.

### LINKED ISSUE #8705: Supporting `<group>` in astropy.io.votable
The VOTable recommendation includes a `<group>` entity (see section 4.9 of [IVOA VOTable](http://www.ivoa.net/documents/latest/VOT.html). This grouping is used by several IVOA recommendation, for instance to specify that a set of columns containing coordinates are related to the same reference frame. As far as I could check, it is not possible to build a VOTable with grouped `param`s or `field_ref`s.

### PR DESCRIPTION
**Pull request by tomdonaldson**
This is a work in progress being discussed at the IVOA interop meeting in Paris - the main goal here is to make sure that additional metadata such as COOSYS is correctly parsed and stored.

this fixes  #7759  and fixes #8705

### CODE REVIEW

**Comment by astrofrog** [comment]
For some reason this is returning an empty coosys for https://drive.google.com/drive/folders/1vzjAs5ANsyfwg3NvYH1GCOj86uT666U5 (Vizier table from 2019)

**Comment by pllim** [comment]
I am not sure if the XML files should be included with the final PR, but rather include a tiny version of the XML that can minimally be used in the test that you would write out of `play.py`. Or maybe just insert metadata in existing test file(s) and test against that/those if they are not already there.

File: astropy/io/votable/tree.py [hunk 1]
```diff
@@ -512,6 +515,11 @@ def parse(self, iterator, config):
     def to_xml(self, w, **kwargs):
         w.element(self._element_name, self._content,
                   attrib=w.object_attrs(self, self._attr_list))
+        
+    def to_meta(self):
+        d = dict([(key, getattr(self, key)) for key in self._attr_list])
```
Comment (pllim) [inline astropy/io/votable/tree.py]: Is it "cleaner" to call parent's `to_meta` here or is that extra overhead? Not sure.

**Comment by tomdonaldson** [comment]
This was a temporary PR to share code for review.  Closing it now.

**Comment by gpdf** [comment]
@tomdonaldson Was this code ultimately merged on some other PR?

**Pull request closed without merging**